packages.config

```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="boost" version="1.65.1.0" targetFramework="native" />
-  <package id="boost_program_options-vc140" version="1.65.1.0" targetFramework="native" />
+  <package id="boost" version="1.66.0.0" targetFramework="native" />
+  <package id="boost_program_options-src" version="1.66.0.0" targetFramework="native" />

```

Comment:
Isn't this package building boost from the sources instead of providing binaries?

Reply:
Yes and it is intended to be that way :). The idea is to have the same package.config for both type of build tools (140_xp (VS 2015) and 141(VS 2017). I agree it increases the compilation time but it will help in the future when VS 15.6 will be out.

Reply:
Got that, not a big fan of pulling in binaries myself, the issue is that building Boost takes forever, hence that hard dependency on 140 in the first place. What build times do you get?

Reply:
PS C:\_2018\february\06\tmp\OpenStackService> Measure-Command {MSBuild.exe /nologo /maxcpucount .\OpenStackService.sln /target:Clean /property:Configuration="SDK10Release" /p:Platform=x64}


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 426
Ticks             : 4260257
TotalDays         : 4.93085300925926E-06
TotalHours        : 0.000118340472222222
TotalMinutes      : 0.00710042833333333
TotalSeconds      : 0.4260257
TotalMilliseconds : 426.0257



PS C:\_2018\february\06\tmp\OpenStackService> Measure-Command {MSBuild.exe /nologo /maxcpucount .\OpenStackService.sln /target:Build /property:Configuration="SDK10Release" /p:Platform=x64}


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 27
Milliseconds      : 404
Ticks             : 274042029
TotalDays         : 0.000317178274305556
TotalHours        : 0.00761227858333333
TotalMinutes      : 0.456736715
TotalSeconds      : 27.4042029
TotalMilliseconds : 27404.2029

Can be tweaked with (/MP):
$ git diff OpenStackService.vcxproj
diff --git a/OpenStackService.vcxproj b/OpenStackService.vcxproj
index a7ce273..cec96df 100644
--- a/OpenStackService.vcxproj
+++ b/OpenStackService.vcxproj
@@ -244,6 +244,16 @@
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='SDK10Release|Win32'">
+    <ClCompile>
+      <MultiProcessorCompilation>true</MultiProcessorCompilation>
+    </ClCompile>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='SDK10Release|x64'">
+    <ClCompile>
+      <MultiProcessorCompilation>true</MultiProcessorCompilation>
+    </ClCompile>
+  </ItemDefinitionGroup>
   <ItemGroup>
     <ClCompile Include="Program.cpp" />
     <ClCompile Include="OpenStackService.cpp" />

to:
PS C:\_2018\february\06\tmp\OpenStackService> Measure-Command {MSBuild.exe /nologo /maxcpucount .\OpenStackService.sln /target:Clean /property:Configuration="SDK10Release" /p:Platform=x64}


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 436
Ticks             : 4363850
TotalDays         : 5.05075231481481E-06
TotalHours        : 0.000121218055555556
TotalMinutes      : 0.00727308333333333
TotalSeconds      : 0.436385
TotalMilliseconds : 436.385



PS C:\_2018\february\06\tmp\OpenStackService> Measure-Command {MSBuild.exe /nologo /maxcpucount .\OpenStackService.sln /target:Build /property:Configuration="SDK10Release" /p:Platform=x64}


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 13
Milliseconds      : 320
Ticks             : 133205666
TotalDays         : 0.000154173224537037
TotalHours        : 0.00370015738888889
TotalMinutes      : 0.222009443333333
TotalSeconds      : 13.3205666
TotalMilliseconds : 13320.5666

Both decent IMO.